packages/mobile/src/components/TokenBottomSheet.tsx

```diff
@@ -0,0 +1,175 @@
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import { useShowOrHideAnimation } from 'src/components/useShowOrHideAnimation'
+import { Namespaces } from 'src/i18n'
+import { useBalance } from 'src/stableToken/hooks'
+import { Currency, STABLE_CURRENCIES } from 'src/utils/currencies'
+
+export enum TokenPickerOrigin {
+  Send = 'Send',
+  Exchange = 'Exchange',
+}
+
+interface Props {
+  isVisible: boolean
+  origin: TokenPickerOrigin
+  onCurrencySelected: (currency: Currency) => void
+}
+
+function CurrencyOption({ currency, onPress }: { currency: Currency; onPress: () => void }) {
+  const { t } = useTranslation(Namespaces.sendFlow7)
+  const balance = useBalance(currency)
+  const amount = {
+    value: new BigNumber(balance ?? '0'),
+    currencyCode: currency,
+  }
+  return (
+    <Touchable onPress={onPress} testID={`${currency}Touchable`}>
+      <View style={styles.currencyOptionContainer}>
+        <Text style={styles.optionName}>{t('stableBalance', { token: currency })}</Text>
+        <View style={styles.currencyBalanceContainer}>
+          <CurrencyDisplay
+            style={styles.localBalance}
+            amount={amount}
+            showLocalAmount={true}
+            testID={`Local${currency}Balance`}
+          />
+          <CurrencyDisplay
+            style={styles.currencyBalance}
+            amount={amount}
+            showLocalAmount={false}
+            hideCode={false}
+            hideSymbol={true}
+            testID={`${currency}Balance`}
+          />
+        </View>
+      </View>
+    </Touchable>
+  )
+}
+
+function TokenBottomSheet({ isVisible, origin, onCurrencySelected }: Props) {
+  const [showingOptions, setOptionsVisible] = useState(isVisible)
+  const [pickerHeight, setPickerHeight] = useState(0)
+
+  const { t } = useTranslation(Namespaces.sendFlow7)
+
+  const onCurrencyPressed = (currency: Currency) => () => {
+    ValoraAnalytics.track(SendEvents.token_selected, {
+      origin,
+      token: currency,
+    })
+    onCurrencySelected(currency)
+  }
+
+  const progress = useSharedValue(0)
+  const animatedPickerPosition = useAnimatedStyle(() => ({
+    transform: [{ translateY: (1 - progress.value) * pickerHeight }],
+  }))
+  const animatedOpacity = useAnimatedStyle(() => ({
+    opacity: 0.5 * progress.value,
+  }))
+
+  useShowOrHideAnimation(
+    progress,
+    isVisible,
+    () => setOptionsVisible(true),
+    () => setOptionsVisible(false)
+  )
+
+  if (!showingOptions) {
+    return null
+  }
+
+  const onLayout = (event: LayoutChangeEvent) => {
+    const { height } = event.nativeEvent.layout
+    setPickerHeight(height)
+  }
+
+  return (
+    <View style={styles.container} testID="TokenBottomSheetContainer">
+      <Animated.View style={[styles.background, animatedOpacity]} />
+      <Animated.View style={[styles.contentContainer, animatedPickerPosition]} onLayout={onLayout}>
```

Comment:
nit: on iOS devices with the home indicator, a 24 bottom margin is not entirely enough, we can do this instead:  https://github.com/celo-org/wallet/blob/b80fceebbd0b9978c33eb6e19520f68b2ed77bcb/packages/mobile/src/verify/VerificationLoadingScreen.tsx#L244

Reply:
Added this fix in https://github.com/celo-org/wallet/pull/346 since it was a lot easier to test in that PR where it was actually implemented